core/src/main/java/fr/sncf/osrd/conflicts/SpacingResourceGenerator.kt

```diff
@@ -229,13 +179,20 @@ class SpacingRequirementAutomaton(
         // more path is needed
         signalLoop@ while (pendingSignals.isNotEmpty()) {
             val pathSignal = pendingSignals.first()
-            if (nextProbedZoneForSignal == -1)
-                nextProbedZoneForSignal = getSignalProtectedZone(pathSignal)
             val physicalSignal = loadedSignalInfra.getPhysicalSignal(pathSignal.signal)
 
             // figure out when the signal is first seen
-            val sightOffset = pathSignal.pathOffset - rawInfra.getSignalSightDistance(physicalSignal)
-            val sightTime = callbacks.arrivalTimeInRange(sightOffset, pathSignal.pathOffset)
+            val signalOffset = incrementalPath.toTravelledPath(pathSignal.pathOffset)
+            val sightOffset = signalOffset - rawInfra.getSignalSightDistance(physicalSignal)
+            // If the train's simulation has reached the point where the signal is seen, bail out
+            if (callbacks.currentPathOffset < sightOffset || callbacks.simulationComplete && incrementalPath.pathComplete)
```

Comment:
The comment + condition don't seem to match. We want to exit if the simulation has *not* reached the sight point.

Exiting if both path and simulation are complete feel wrong, it means we don't process any signal if the whole path is known from the start.

Reply:
you're indeed right, the tests didn't pass because of it